packages/story-editor/src/components/panels/design/sizePosition/radius.js

```diff
@@ -185,87 +121,14 @@ function RadiusControls({ selectedElements, pushUpdateForObject }) {
-          placeholder={
-            borderRadius.topLeft === MULTIPLE_VALUE
-              ? MULTIPLE_DISPLAY_VALUE
-              : ''
-          }
-          isIndeterminate={borderRadius.topLeft === MULTIPLE_VALUE}
-          containerStyleOverride={
-            lockRadius ? inputContainerStyleOverride : styleOverrideTopLeft
-          }
-        />
-        {!lockRadius && (
-          <>
-            <BoxedNumericInput
-              isSmall
-              corner="topRight"
-              suffix={<Icons.Corner />}
-              value={
-                borderRadius.topRight === MULTIPLE_VALUE
-                  ? ''
-                  : borderRadius.topRight
-              }
-              aria-label={__('Top right corner radius', 'web-stories')}
-              onChange={(_, value) => handleChange('topRight', value)}
-              placeholder={
-                borderRadius.topRight === MULTIPLE_VALUE
-                  ? MULTIPLE_DISPLAY_VALUE
-                  : ''
-              }
-              isIndeterminate={borderRadius.topRight === MULTIPLE_VALUE}
-              containerStyleOverride={styleOverrideTopRight}
-            />
-            <BoxedNumericInput
-              isSmall
-              value={
-                borderRadius.bottomLeft === MULTIPLE_VALUE
-                  ? ''
-                  : borderRadius.bottomLeft
-              }
-              aria-label={__('Bottom left corner radius', 'web-stories')}
-              onChange={(_, value) => handleChange('bottomLeft', value)}
-              placeholder={
-                borderRadius.bottomLeft === MULTIPLE_VALUE
-                  ? MULTIPLE_DISPLAY_VALUE
-                  : ''
-              }
-              suffix={<Icons.Corner />}
-              isIndeterminate={borderRadius.bottomLeft === MULTIPLE_VALUE}
-              containerStyleOverride={styleOverrideBottomLeft}
-            />
-            <BoxedNumericInput
-              isSmall
-              value={
-                borderRadius.bottomRight === MULTIPLE_VALUE
-                  ? ''
-                  : borderRadius.bottomRight
-              }
-              aria-label={__('Bottom right corner radius', 'web-stories')}
-              onChange={(_, value) => handleChange('bottomRight', value)}
-              placeholder={
-                borderRadius.bottomRight === MULTIPLE_VALUE
-                  ? MULTIPLE_DISPLAY_VALUE
-                  : ''
-              }
-              corner="bottomRight"
-              suffix={<Icons.Corner />}
-              isIndeterminate={borderRadius.bottomRight === MULTIPLE_VALUE}
-              containerStyleOverride={styleOverrideBottomRight}
-            />
-          </>
-        )}
-      </InputContainer>
+      <StackedInputs
+        lockInput={lockRadius}
+        inputProps={borderRadius}
+        handleChange={handleChange}
+        suffix={<Icons.Corner />}
+        firstInputLabel={firstInputLabel}
+        showLockedSuffixIcon
+      />
```

Comment:
I do see how this cleans up this component a lot and I do appreciate that. But I also think it creates a new component super-specific to only these two instances, that can't be used anywhere else ever again, and thus isn't that relevant in the codebase.

Reply:
There might be room for a `<LockableStackableInputs />` component somewhere though, that puts the lock inside the components and you could then use conditional rendering of the children object to only render the first input rather than all children in case of locked/unlocked inputs - this might even be reusable for the text box padding inputs actually. But that's a rather specific component, that is a bit much work I think.

Reply:
Agreed with Morten here that the new component is currently specific only to these two use cases and doesn't make much sense as a generic form component. Any generic form component should support new use cases and only include the shared parts of the input(s). Labels, icons, etc. would all come from the specific use case and be outside of the generic form component.